### PR TITLE
Added a new datasource google_sql_database_instance_latest_recovery_time

### PR DESCRIPTION
**Pull request by ibhaskar2**
A new datasource google_sql_database_instance_latest_recovery_time for fetching the latest recovery time of csql instance for PITR. It receives instance as argument and returns latest recovery timestamp.

I have added documentation and tested this new data source with unit tests.

b/276255004

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-datasource
`google_sql_database_instance_latest_recovery_time`
```

### CODE REVIEW

File: mmv1/third_party/terraform/website/docs/d/sql_database_instance_latest_recovery_time.html.markdown [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+* `instance` - (Required) The name of the instance.
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `instance` - The name of the instance.
+* `project` - The ID of the project in which the resource belongs.
+* `tiers` - Timestamp, identifies the latest recovery time of the source instance.
```
Comment (zli82016) [inline mmv1/third_party/terraform/website/docs/d/sql_database_instance_latest_recovery_time.html.markdown:39]: Is `tiers` exported?

File: mmv1/third_party/terraform/tests/data_source_sql_database_instance_latest_recovery_time_test.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
+package google
+
```
Comment (zli82016) [inline mmv1/third_party/terraform/tests/data_source_sql_database_instance_latest_recovery_time_test.go:2]: Can you please move this file to services/sql folder, the same folder with the data source? And the package name should be `sql_test`?

File: mmv1/third_party/terraform/tests/data_source_sql_database_instance_latest_recovery_time_test.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      start_time                     = "20:55"
+      transaction_log_retention_days = "3"
+    }
+  }
+
+  deletion_protection = false
+}
+
+data "google_sql_database_instance_latest_recovery_time" "default" {
+	instance = resource.google_sql_database_instance.main.name
```
Comment (zli82016) [inline mmv1/third_party/terraform/tests/data_source_sql_database_instance_latest_recovery_time_test.go:56]: Can you modify the indentation here?

**Review by zli82016 — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/website/docs/d/sql_database_instance_latest_recovery_time.html.markdown (hunk 1 above)
Comment (ibhaskar2) [reply mmv1/third_party/terraform/website/docs/d/sql_database_instance_latest_recovery_time.html.markdown]: Corrected it.

File: mmv1/third_party/terraform/tests/data_source_sql_database_instance_latest_recovery_time_test.go (hunk 2 above)
Comment (ibhaskar2) [reply mmv1/third_party/terraform/tests/data_source_sql_database_instance_latest_recovery_time_test.go]: Done

File: mmv1/third_party/terraform/tests/data_source_sql_database_instance_latest_recovery_time_test.go (hunk 3 above)
Comment (ibhaskar2) [reply mmv1/third_party/terraform/tests/data_source_sql_database_instance_latest_recovery_time_test.go]: Done

**Review by zli82016 — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**